bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/Util.java

```diff
@@ -63,12 +62,41 @@ public static String requestToLogString(Request request) {
      * @throws URISyntaxException if parameter could not be converted to a URI
      */
     public static URI uriFromString(String s) throws MalformedURLException, URISyntaxException {
-        URL url = new URL(s);
-        URI uri = new URI(url.getProtocol(), url.getUserInfo(), IDN.toASCII(url.getHost()), url.getPort(),
-                url.getPath(), url.getQuery(), url.getRef());
+        URI uri = parse(s);
         return URI.create(uri.toASCIIString().replace("+", "%2B").replace("%25%25", "%"));
     }
 
+    private static final Pattern URL_PATTERN = Pattern.compile("^(?:(?<scheme>[a-zA-Z][a-zA-Z0-9+.-]*):)?//?"
+            + "(?:(?<userinfo>[^@/?#]*)@)?" + "(?<host>[^:/?#]*)(?::(?<port>\\d+))?" + "(?<path>/[^?#]*)?"
```

Comment:
Should IPv6 addresses be allowed? [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986) includes them, wrapped in square brackets, but they would fail validation here since they use `:` as separator.

Reply:
```suggestion
            + "(?:(?<userinfo>[^@/?#]*)@)?" + "(?<host>(?:\\[[a-fA-F0-9:]+\\])|(?:[^:/?#]*))(?::(?<port>\\d+))?" + "(?<path>/[^?#]*)?"
```

Reply:
@pacive Could you double check this is ok?